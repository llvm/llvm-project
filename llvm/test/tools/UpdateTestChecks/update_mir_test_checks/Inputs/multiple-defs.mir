# RUN: llc -mtriple=x86_64 -run-pass=none -verify-machineinstrs %s -o - | FileCheck %s

---
name: test
body: |
  bb.0:
    %0:gr32 = IMPLICIT_DEF
    %1:gr32 = IMPLICIT_DEF
    %0:gr32 = IMPLICIT_DEF
    %2:gr32 = IMPLICIT_DEF
    KILL %0, %2
...
