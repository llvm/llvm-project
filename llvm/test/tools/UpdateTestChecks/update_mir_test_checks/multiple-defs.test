# REQUIRES: x86-registered-target
## Check that update_mir_test_checks handles multiple definitions of the same
## virtual register (after coming out of SSA form).

# RUN: cp -f %S/Inputs/multiple-defs.mir %t.mir && %update_mir_test_checks %t.mir
# RUN: diff -u %S/Inputs/multiple-defs.mir.expected %t.mir
# RUN: FileCheck %t.mir < %t.mir
