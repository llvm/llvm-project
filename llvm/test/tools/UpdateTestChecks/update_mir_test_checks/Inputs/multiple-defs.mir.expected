# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
# RUN: llc -mtriple=x86_64 -run-pass=none -verify-machineinstrs %s -o - | FileCheck %s

---
name: test
body: |
  bb.0:
    ; CHECK-LABEL: name: test
    ; CHECK: [[DEF:%[0-9]+]]:gr32 = IMPLICIT_DEF
    ; CHECK-NEXT: [[DEF1:%[0-9]+]]:gr32 = IMPLICIT_DEF
    ; CHECK-NEXT: [[DEF:%[0-9]+]]:gr32 = IMPLICIT_DEF
    ; CHECK-NEXT: [[DEF2:%[0-9]+]]:gr32 = IMPLICIT_DEF
    ; CHECK-NEXT: KILL [[DEF]], [[DEF2]]
    %0:gr32 = IMPLICIT_DEF
    %1:gr32 = IMPLICIT_DEF
    %0:gr32 = IMPLICIT_DEF
    %2:gr32 = IMPLICIT_DEF
    KILL %0, %2
...
